orca-webhook/src/main/groovy/com/netflix/spinnaker/orca/webhook/tasks/MonitorWebhookTask.groovy

```diff
@@ -49,15 +48,21 @@ class MonitorWebhookTask implements Task {
     String successStatuses = stage.context.successStatuses
     String canceledStatuses = stage.context.canceledStatuses
     String terminalStatuses = stage.context.terminalStatuses
+    List<Map<String, String>> headers = stage.context.headers ? (stage.context.headers as List<Map<String, String>>) : []
+
+    def response = webhookService.getStatus(statusEndpoint, headers)
 
-    def response = webhookService.getStatus(statusEndpoint)
     def result
     try {
       result = JsonPath.read(response.body, statusJsonPath)
     } catch (PathNotFoundException e) {
       return new TaskResult(ExecutionStatus.TERMINAL,
         [error: [reason: e.message, response: response.body]])
     }
+
+    if (isSingleElementCollection(result))
```

Comment:
Support for collection with single element, to handle JSONPath returning list as response